feat(LIVE-6958): replace ProviderIcons with svg hosted from CDN

cng-ledger opened the pull request:
<!--
Thank you for your contribution! 👍
Please make sure to read CONTRIBUTING.md if you have not already.
Disclaimer: Pull Requests that do not comply with the rules will be arbitrarily closed.
-->

### 📝 Description

- Remove existing `ProviderIcon` reference from `@ledgerhq/react-ui`
- Added CDN url, Provider icon url helper & Provider icon sizes to `@ledgerhq/live-common`
- Added `@ledgerhq/ui-shared` as a dependency to both LLD & LLM
- Added `ProviderIcon` component to LLD & LLM which loads icon from CDN

to support loading Provider's icon using SVG hosted from CDN

### ❓ Context

- **Impacted projects**: `LLD` <!-- The list of end user projects impacted by the change. -->
- **Linked resource(s)**: `https://ledgerhq.atlassian.net/browse/LIVE-6958` <!-- Attach any ticket number if relevant. (JIRA / Github issue number) -->

### ✅ Checklist

- [ ] **Test coverage** <!-- Are your changes covered by tests? Features must be tested, bugfixes must include a test that would have detected the issue. -->
- [ ] **Atomic delivery** <!-- Is this pull request standalone? In order words, does it depend on nothing else? Please explain if not checked. -->
- [ ] **No breaking changes** <!-- If there are breaking changes, please explain why. -->

### 📸 Demo

<!--
For visual features, please attach screenshots or video recordings to demonstrate the changes.
For libraries, you can add a code sample.
For bugfixes, you can drop this section.
-->

### 🚀 Expectations to reach

_Please make sure you follow these [**Important Steps**](https://github.com/LedgerHQ/ledger-live/blob/develop/CONTRIBUTING.md#important-steps)._

_Pull Requests must pass the CI and be internally validated in order to be merged._

<!-- If any of the expectations are not met please explain the reason in detail. -->

sarneijim commented on libs/ui/packages/shared/src/consts/urls.ts:
```diff
@@ -0,0 +1 @@
+export const CDN_URL = 'https://cdn.live.ledger.com';
```
The idea behind this pr is remove providers icons library from react-ui. The icons will not be longer imported from here.

cng-ledger replied:
I've reworked this PR. Removed all references of `ProviderIcon` from `@ledgerhq/react-ui`. Replaced them with a newly added `ProviderIcon` in the LLD / LLM app

sarneijim commented on apps/ledger-live-desktop/src/renderer/screens/exchange/Swap2/Form/Rates/Rate.tsx:
```diff
@@ -1,9 +1,9 @@
 import Box from "~/renderer/components/Box";
 import { ExchangeRate } from "@ledgerhq/live-common/exchange/swap/types";
-import { iconByProviderName } from "../../utils";
+
```
Sorry, but perhaps the jira was not clear enough. We need to remove any ProviderIcon  dependency from @ledgerhq/react-ui

cng-ledger replied:
Similar to comment above. I've reworked this, however it does still load asset from CDN just not from `@ledgerhq/react-ui`.

sarneijim reviewed (commented):
Hello, congrats for your first pr!
https://media.giphy.com/media/xT8qBepJQzUjXpeWU8/giphy.gif

Do not forget we have LLM and LLD. I do not mind if we use 2 different pr for them, but in that case, ca you create a 2 subtasks?

cng-ledger commented:
> Hello, congrats for your first pr! https://media.giphy.com/media/xT8qBepJQzUjXpeWU8/giphy.gif
> 
> Do not forget we have LLM and LLD. I do not mind if we use 2 different pr for them, but in that case, ca you create a 2 subtasks?

Rewrote this PR, added code change to both LLD & LLM app.

kallen-ledger reviewed (commented):
Looks good, just one small comment

sarneijim commented on libs/ui/packages/shared/src/assorted/Icon/Providers.ts:
```diff
@@ -0,0 +1,56 @@
+type GetProviderIconUrlProps = {
+  cdn?: string;
+  name: Provider;
+  boxed: boolean;
```
Checking your logic I think this is optional
```suggestion
  boxed?: boolean;
```

sarneijim commented on libs/ui/packages/shared/src/assorted/Icon/Providers.ts:
```diff
@@ -0,0 +1,56 @@
+export type Provider = (typeof ALL_PROVIDERS)[number];
+
+export const isProviderIconName = (name: string): name is Provider =>
+  ALL_PROVIDERS.some((item) => item === name);
```
I am not sure about this check...
For instance, ETH stake banner is controlled by a FF, if Product change the FF data, the banner will change the appearance.
If we need to add a new provider to list, we will need to plan the change with 3 weeks in advance (LLM and LLD releases)
Can we avoid this check?
```
{"enabled":true,"params":{"listProvider":[{"id":"kiln_pooling","liveAppId":"kiln","supportLink":"https://www.kiln.fi","icon":"Kiln:provider","queryParams":{"focus":"pooled"}},{"id":"kiln","liveAppId":"kiln","supportLink":"https://www.kiln.fi","icon":"Kiln:provider","queryParams":{"focus":"dedicated"}}]}}
```

cng-ledger replied:
List of supported providers, icon name validation has been removed to keep this more flexible

sarneijim reviewed (commented):
.

mle-gall commented:
Good to see this coming finally ! Thanks a lot @sarneijim 🔥

mle-gall commented on libs/ui/packages/shared/index.ts:
```diff
@@ -1 +1,4 @@
+export * from "./src/assorted/Icon/Providers";
+export * from "./src/assorted/Icon/Sizes";
+export * from "./src/consts/urls";
```
Please don't put the provider icon logic in UI-Shared as it's reserved for the Design System. We want to get rid of the Provider Icons in the Design System. UI Shared is reserved for design system design tokens.
I'd recommend to have that in LLC instead of UI-Shared.

cng-ledger replied:
@mle-gall Thanks for the feedback. I've moved this code to LLC instead of UI-shared

sarneijim reviewed (changes requested):
Unable to run pnpm dev:lld after installation. Can you?
<img width="1017" alt="image" src="https://github.com/LedgerHQ/ledger-live/assets/38540290/68fadc32-f711-434b-a5ca-635f97d935a0">

sarneijim commented:
Do not forget add the changeset

sarneijim commented on libs/ledger-live-common/src/icons/providers/providers.ts:
```diff
@@ -0,0 +1,12 @@
+type GetProviderIconUrlProps = {
+  cdn?: string;
+  name: string;
+  boxed: boolean;
```
could this param be optional?

cng-ledger replied:
If I change `boxed` in `getProviderIconUrl` to optional. Can I remove the default value in the `ProviderIcon/index.tsx` in both LLD & LLM?

sarneijim commented:
/generate-screenshots

ggilchrist-ledger commented:
/generate-screenshots

cng-ledger commented:
/generate-screenshots

cng-ledger commented:
/generate-screenshots

mle-gall reviewed (approved):
Seems all good on Live Hub side 🙏

sarneijim reviewed (commented):
I am rerunning the screenshots as we are not seeing correctly the the kiln icon. In case a new screenshot will not solve the issue, please be sure you can see Kiln correctly before merged.

cng-ledger commented:
@sarneijim It seemed okay when i run it locally 🤔

https://cdn.live.ledger.com/icons/providers/boxed/kiln.svg

<img width="705" alt="Screenshot 2023-07-19 at 15 15 55" src="https://github.com/LedgerHQ/ledger-live/assets/138497251/2588fdca-8132-4ae9-8d16-3cf1f3c5afaa">

cng-ledger commented:
Is weird `live-common-tools` deployment step to Vercel fails because `pnpm turbo run build --filter="live-common-tools"` works locally
